python/adjoint/objective.py

```diff
@@ -92,6 +92,8 @@ def _adj_src_scale(self, include_resolution=True):
         else:
             # multi frequency simulations
             scale = dV * iomega / adj_src_phase
+        if ((not self.sim.k_point) or self.sim.k_point==mp.Vector3()) and not self.sim.force_complex_fields:
```

Comment:
We should really check `fields.is_real`, assuming the `fields` object exists here.

Reply:
Even simpler, check `self.use_real()`

Reply:
`use_real` is currently an internal function of `init_sim`, which can just be split into a separate method, and probably renamed to `using_real_fields()`